library_generation/utils/pom_generator.py

```diff
@@ -0,0 +1,157 @@
+    render(
+        template_name="root-pom.xml.j2",
+        output_name=f"{repository_path}/pom.xml",
+        modules=modules,
+    )
+
+
+def generate_gapic_bom(repository_path: str, versions_file: str) -> None:
+    print("Regenerating gapic-libraries-bom")
+    bom_configs = __search_for_bom_artifact(repository_path)
+    monorepo_version = get_version_from(
+        versions_file=versions_file,
+        artifact_id="google-cloud-java",
+    )
+    render(
+        template_name="gapic-libraries-bom.xml.j2",
+        output_name=f"{repository_path}/gapic-libraries-bom/pom.xml",
+        monorepo_version=monorepo_version,
+        bom_configs=bom_configs,
+    )
+
+
+def get_version_from(
+    versions_file: str, artifact_id: str, is_released: bool = False
+) -> str:
+    """
+    Get version of a given artifact from versions.txt
+    :param versions_file: the path of version.txt
+    :param artifact_id: the artifact id
+    :param is_released: whether returns the released or current version
+    :return: the version of the artifact
+    """
+    index = 1 if is_released else 2
+    with open(versions_file, "r") as f:
+        for line in f.readlines():
+            if artifact_id in line:
+                return line.split(":")[index].strip()
+
+
+def __search_for_java_modules(
+    repository_path: str,
+) -> List[str]:
+    repo = Path(repository_path).resolve()
+    modules = []
+    for sub_dir in repo.iterdir():
+        if sub_dir.is_dir() and sub_dir.name.startswith("java-"):
+            modules.append(sub_dir.name)
+    return sorted(modules)
+
+
+def __search_for_bom_artifact(
+    repository_path: str,
+) -> List[BomConfig]:
+    repo = Path(repository_path).resolve()
+    module_exclusions = ["gapic-libraries-bom"]
+    group_id_inclusions = [
+        "com.google.cloud",
+        "com.google.analytics",
+        "com.google.area120",
+    ]
+    bom_configs = []
+    for module in repo.iterdir():
+        if module.is_file() or module.name in module_exclusions:
+            continue
+        for sub_module in module.iterdir():
+            if sub_module.is_dir() and sub_module.name.endswith("-bom"):
+                root = etree.parse(f"{sub_module}/pom.xml").getroot()
+                group_id = root.find(f"{project_tag}{group_id_tag}").text
+                if group_id not in group_id_inclusions:
+                    continue
+                artifact_id = root.find(f"{project_tag}{artifact_tag}").text
+                version = root.find(f"{project_tag}{version_tag}").text
+                index = artifact_id.rfind("-")
+                version_annotation = artifact_id[:index]
+                bom_configs.append(
+                    BomConfig(
+                        group_id=group_id,
+                        artifact_id=artifact_id,
+                        version=version,
+                        version_annotation=version_annotation,
+                    )
+                )
+    # handle edge case: java-grafeas
```

Comment:
I know this was done in previous PRs, what is special about these three modules?

Reply:
These three modules are not generated any more.

However, these modules are still part of the pom, so I need to add them using different approach since they don't have similar structure as other modules.

Reply:
I see, do we have to do the same for every handwritten libraries in `google-cloud-java`?

Reply:
No. I don't think this step is needed in handwritten libraries.

We need to refactor to only applicable in monorepo.